design-guidelines/form-ux.md

```diff
@@ -0,0 +1,200 @@
+
+## Structure and hierarchy
+
+Forms are designed with a consistent hierarchy and structure to make it easier for users to scan the inputs and submit their data.
+
+Every form will have at least one section, but form sections should always be used when a complex form can be broken down into distinct sections of related fields.
+
+For more information about the parts that make up a form, see the Storybook documentation for each component under the "Form structure" section.
+
+### Field order
+
+Order fields by their relative level of importance. The inputs that are most critical for the user to complete their task should come first.
+
+### Field grouping
+
+Fields that are closely related should be visually grouped together.
+
+For more information on grouped fields, see the Storybook documentation for the `FieldList` and `FieldGroup` components.
+
+### Form actions
+
+A form should contain at least two buttons: one to submit the form, and one to navigate away from the form without submitting any of the input.
+
+These buttons should be highly visible, and easy for the user to press.
+
+### Multi-part forms
+
+Forms should be kept as succinct as possible, but some forms require a lot of data input. Break complex forms into multiple forms when it will help the user focus on what data they're inputting.
+
+#### Tabbed form pattern
+
+Break a form into different tabs when its fields don't fit on one screen without overwhelming the user.
+
+Keep in mind that breaking a form into separate tabs forces the user to navigate to find a specific field. Only use this pattern if tabbed sections make it easier to find a specific field.
+
+#### Progressive disclosure pattern
+
+If a complex form can be broken into a linear series of steps, sections can be progressively shown as the user inputs their data.
+
+A progressive disclosure pattern is also useful when the value of one field will affect subsequent fields. For example: selecting a cloud provider before showing the fields for authorization credentials.
+
+## Kinds of inputs
+
+### Text inputs
+
+Use text inputs when the field value is expected to be free-form text, and we can't provide a set of possible values for a user to choose from. For example: setting a cluster's display name.
+
+For more detailed information on text inputs, see the Storybook documentation for the `TextInput` and `Textarea` components.
+
+### Choice inputs
+
+Use a choice input when a field has a known set of possible values. For example: selecting which version of Kubernetes should be used when creating a cluster.
+
+Some examples of choice inputs include: typeahead dropdowns, `<select>` dropdowns, checkboxes, file upload fields, and segmented controls.
+
+For more detailed information on choice inputs, see the respective Storybook documentation for the respective choice input components (e.g.: `CheckboxInput`, `SegmentedControl`, `SelectInput`, `ToggleBox`, etc)
+
+## Field parts
```

Comment:
I think that might be more confusing because a field contains more than just form elements. It also includes a label, help text, and/or validation text.

That could just be me though.